src/runtime/pkg/kata-monitor/cri.go

```diff
@@ -132,29 +131,35 @@ func (km *KataMonitor) getSandboxes(sandboxMap map[string]bool) (map[string]bool
 	request := &pb.ListPodSandboxRequest{
 		Filter: filter,
 	}
-	monitorLog.Debugf("ListPodSandboxRequest: %v", request)
+	monitorLog.Tracef("ListPodSandboxRequest: %v", request)
 	r, err := runtimeClient.ListPodSandbox(context.Background(), request)
 	if err != nil {
-		return newMap, err
+		return sandboxList, err
 	}
-	monitorLog.Debugf("ListPodSandboxResponse: %v", r)
+	monitorLog.Tracef("ListPodSandboxResponse: %v", r)
 
 	for _, pod := range r.Items {
-		// Use the cached data if available
-		if isKata, ok := sandboxMap[pod.Id]; ok {
-			newMap[pod.Id] = isKata
-			continue
+		for _, sandbox := range sandboxList {
+			if pod.Id == sandbox {
+				km.sandboxCache.setMetadata(sandbox, sandboxKubeData{
+					uid:       pod.Metadata.Uid,
+					name:      pod.Metadata.Name,
+					namespace: pod.Metadata.Namespace,
+				})
+
+				sandboxList = removeFromSandboxList(sandboxList, sandbox)
+
+				monitorLog.WithFields(logrus.Fields{
+					"Pod Name":      pod.Metadata.Name,
+					"Pod Namespace": pod.Metadata.Namespace,
+					"Pod UID":       pod.Metadata.Uid,
+				}).Debugf("Synced KATA POD %s", pod.Id)
```

Comment:
```suggestion
				monitorLog.WithFields(logrus.Fields{
					"Pod Name":      pod.Metadata.Name,
					"Pod Namespace": pod.Metadata.Namespace,
					"Pod UID":       pod.Metadata.Uid,
					"Pod ID":         pod.Id,
				}).Debug("Synced pod")
```

But also, it's much easier for consumers of these logs if we avoid spaces in fields (and use dashes instead).

Reply:
Yep, I will fix this in a follow-up PR. Regarding #3705, while here we are still enforcing CRI , now that we basically do detection based on FS events only, we can make CRI optional. Something I will take care in a follow-up PR soon.